### PR TITLE
static abstract interface members

### PR DESCRIPTION
**Pull request by Natalia-77**
This branch represent static abstract interface members in generic type

### CODE REVIEW

**Review by vlada-shubina — COMMENTED** [review]
it would be good to add tests to showcase results.
`RomanNumericNumber`:
- add `ToString() showing the number in roman notation
- add property that can return number in roman notation

File: CalculatorOperations.UnitTests/RomanTests.cs [hunk 1]
```diff
@@ -14,12 +14,14 @@ public void CheckResultParseRomanToAradNumber(string roman, int arab)
...
-        [InlineData("IV", "4")]
-        public void ShowRomanToAradNumber(string roman, string expected)
+        [InlineData("V", "V")]
+        public void ShowRomanToAradNumber(string roman, string expectedResult)
         {
+            string res = string.Empty;
             var parsedRomanNumeric = RomanNumericsNumber.Parse(roman, null);
-            var parsedNumberToString = parsedRomanNumeric.ToString();
-            Assert.Equal(expected, parsedNumberToString);
+            parsedRomanNumeric.NumberRoman = res;
```
Comment (Natalia-77) [inline CalculatorOperations.UnitTests/RomanTests.cs]: not sure this line.Does it correct?I don't know where I must to call this property.If doesn't call this prop-it"s null.

File: Calculator.Generic.INumber/RomanNumericsNumber.cs [hunk 2]
```diff
@@ -19,6 +18,7 @@ public class RomanNumericsNumber : IAdditionOperators<RomanNumericsNumber, Roman
 
         });
         private int _numerics;
+        private string _numberRoman;
```
Comment (vlada-shubina) [inline Calculator.Generic.INumber/RomanNumericsNumber.cs:21]: this variable is never set, should be set in constructor when creating the instance.

File: Calculator.Generic.INumber/RomanNumericsNumber.cs [hunk 3]
```diff
@@ -28,9 +28,25 @@ public int Numerics
             get => _numerics;
             init => _numerics = value < 0 ? throw new ArgumentException("Operand should be positive", nameof(value)) : value;
         }
+
+        public string NumberRoman
+        {
+            get => _numberRoman;
+            set
```
Comment (vlada-shubina) [inline Calculator.Generic.INumber/RomanNumericsNumber.cs]: logically, it should be read-only so no setter is needed.

File: Calculator.Generic.INumber/RomanNumericsNumber.cs [hunk 4]
```diff
@@ -63,33 +65,42 @@ public override string ToString()
...
-            for (int i = 0; i <= len - 1; i++)
+            var resultArabicNumber = 0;
+            var count = 0;
+            var len = strToParse.Length;
+            for (var i = 0; i <= len - 1; i++)
             {
-                var current = _dictionaryValues[ strToParse[ i ].ToString() ];
-                count++;
-                if (count < len)
+                var isExist = dictionaryValues.ContainsKey(strToParse[ i ].ToString());
```
Comment (vlada-shubina) [inline Calculator.Generic.INumber/RomanNumericsNumber.cs:78]:
```suggestion
                var exists = dictionaryValues.ContainsKey(strToParse[ i ].ToString());
```

File: Calculator.Generic.INumber/RomanNumericsNumber.cs [hunk 5]
```diff
@@ -63,33 +65,42 @@ public override string ToString()
...
                     else
                     {
                         resultArabicNumber += current;
                     }
+
                 }
                 else
                 {
-                    resultArabicNumber += current;
+                    throw new ArgumentException("There is no such key");
```
Comment (vlada-shubina) [inline Calculator.Generic.INumber/RomanNumericsNumber.cs:103]: please do a clear message here. User won't understand this.
Something like: Failed to parse value 'X' as roman, symbol 'Y' is not a valid roman literal.

**Pull request merged**